src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/Runners/AppRunner.cs

```diff
@@ -0,0 +1,178 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests.Runners
+{
+    /// <summary>
+    /// Runner for running the unit test application.
+    /// </summary>
+    internal sealed class AppRunner : BaseRunner
+    {
+        private TaskCompletionSource<object> _currentCommandSource;
+
+        /// <summary>
+        /// The path of the currently executing assembly.
+        /// </summary>
+        private static string CurrentExecutingAssemblyPath =>
+            Assembly.GetExecutingAssembly().Location;
+
+        /// <summary>
+        /// The path to the application.
+        /// </summary>
+        private static string AppPath =>
+            CurrentExecutingAssemblyPath
+                .Replace(Assembly.GetExecutingAssembly().GetName().Name, "Microsoft.Diagnostics.Monitoring.UnitTestApp");
+
+        /// <summary>
+        /// The mode of the diagnostic port connection. Default is <see cref="DiagnosticPortConnectionMode.Listen"/>
+        /// (the application is listening for connections).
+        /// </summary>
+        /// <remarks>
+        /// Set to <see cref="DiagnosticPortConnectionMode.Connect"/> if app needs to connect to a diagnostic port listener.
+        /// </remarks>
+        public DiagnosticPortConnectionMode ConnectionMode { get; set; } = DiagnosticPortConnectionMode.Listen;
+
+        /// <summary>
+        /// Path of the diagnostic port to connect to when <see cref="ConnectionMode"/> is <see cref="DiagnosticPortConnectionMode.Connect"/>.
+        /// </summary>
+        public string DiagnosticPortPath { get; set; }
+
+        /// <summary>
+        /// Name of the scenario to run in the application.
+        /// </summary>
+        public string ScenarioName { get; set; }
+
+        public AppRunner(ITestOutputHelper outputHelper, int appId = 1)
+            : base(CreateRunnerOptions(outputHelper, appId))
+        {
+        }
+
+        protected override IEnumerable<string> GetProcessArguments()
+        {
+            List<string> argsList = new();
+
+            argsList.Add(ScenarioName);
+
+            return argsList;
+        }
+
+        protected override IDictionary<string, string> GetProcessEnvironment()
+        {
+            IDictionary<string, string> environment = base.GetProcessEnvironment();
+
+            if (ConnectionMode == DiagnosticPortConnectionMode.Connect)
+            {
+                if (string.IsNullOrEmpty(DiagnosticPortPath))
+                {
+                    throw new ArgumentNullException(nameof(DiagnosticPortPath));
+                }
+
+                environment.Add("DOTNET_DiagnosticPorts", DiagnosticPortPath);
+            }
+
+            return environment;
+        }
+
+        protected override void OnStandardOutputLine(string line)
+        {
```

Comment:
As mentioned before, we should also be logging stderr

Reply:
Fixed